extensions-contrib/graphite-emitter/src/main/java/io/druid/emitter/graphite/GraphiteEmitter.java

```diff
@@ -174,12 +172,16 @@ public void run()
             log.error(e, e.getMessage());
             if (e instanceof InterruptedException) {
               Thread.currentThread().interrupt();
+              break;
             } else if (e instanceof SocketException) {
+              // This is antagonistic to general Closeable contract in Java,
+              // it is needed to allow re-connection in case of the socket is closed due long period of inactivity
+              pickledGraphite.close();
+              log.warn("Trying to re-connect to graphite server");
               pickledGraphite.connect();
             }
           }
         }
-        pickledGraphite.flush();
```

Comment:
@b-slim - why did you remove `flush()`?
I might be wrong, but seems to me that if you have some metrics in `pickledGraphite` that weren't written to the OutputStream in the `while` loop, they will never be sent (as `pickledGraphite.send()` only writes to the OutputStream when the size of the metrics' list is equal or greater than the batch size, see `PickledGraphite.send()`)